src/main/java/org/spongepowered/api/data/key/Key.java

```diff
@@ -49,6 +50,14 @@
     Class<V> getValueClass();
 
     /**
+     * Gets the class of the element of the {@link Value} this {@link Key}
+     * is representing.
+     *
+     * @return The element class
+     */
+    Class<?> getElementClass();
```

Comment:
Wouldn't it be more usefull to provide a `TypeToken` instead of a `Class`, this could contain the generics of `Map`s, `Set`s and can be used to get the serializer of the values (and keys).